src/file-table-engine/src/error.rs

```diff
@@ -0,0 +1,160 @@
+    CheckObject {
+        path: String,
+        location: Location,
+        source: object_store::Error,
+    },
+
+    #[snafu(display("Fail to encode object into json , source: {}", source))]
+    EncodeJson {
+        location: Location,
+        source: JsonError,
+    },
+
+    #[snafu(display("Fail to decode object from json , source: {}", source))]
+    DecodeJson {
+        location: Location,
+        source: JsonError,
+    },
+
+    #[snafu(display("Failed to drop table, table :{}, source: {}", table_name, source,))]
+    DropTable {
+        source: BoxedError,
+        table_name: String,
+        location: Location,
+    },
+
+    #[snafu(display(
+        "Failed to write table manifest,  table: {}, source: {}",
+        table_name,
+        source,
+    ))]
+    WriteTableManifest {
+        source: object_store::Error,
+        table_name: String,
+        location: Location,
+    },
+
+    #[snafu(display("Failed to write immutable manifest, path: {}", path))]
+    WriteImmutableManifest { path: String, location: Location },
+
+    #[snafu(display("Failed to delete table table manifest, source: {}", source,))]
+    DeleteTableManifest {
+        source: object_store::Error,
+        table_name: String,
+        location: Location,
+    },
+
+    #[snafu(display(
+        "Failed to read table manifest,  table: {}, source: {}",
+        table_name,
+        source,
+    ))]
+    ReadTableManifest {
+        source: object_store::Error,
+        table_name: String,
+        location: Location,
+    },
+
+    #[snafu(display(
+        "Failed to build table meta for table: {}, source: {}",
+        table_name,
+        source
+    ))]
+    BuildTableMeta {
+        source: TableMetaBuilderError,
+        table_name: String,
+        location: Location,
+    },
+
+    #[snafu(display(
+        "Failed to build table info for table: {}, source: {}",
+        table_name,
+        source
+    ))]
+    BuildTableInfo {
+        source: TableInfoBuilderError,
+        table_name: String,
+        location: Location,
+    },
+
+    #[snafu(display("Table already exists: {}", table_name))]
+    TableExists {
+        location: Location,
+        table_name: String,
+    },
+
+    #[snafu(display(
+        "Failed to convert metadata from deserialized data, source: {}",
+        source
+    ))]
+    ConvertRaw {
+        #[snafu(backtrace)]
+        source: table::metadata::ConvertError,
+    },
+
+    #[snafu(display("Invalid schema, source: {}", source))]
+    InvalidRawSchema {
+        #[snafu(backtrace)]
+        source: datatypes::error::Error,
+    },
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+impl ErrorExt for Error {
+    fn status_code(&self) -> StatusCode {
+        use Error::*;
+
+        match self {
+            TableExists { .. }
+            | BuildTableMeta { .. }
+            | BuildTableInfo { .. }
+            | InvalidRawSchema { .. }
```

Comment:
`EncodeJson` and `DecodeJson` are not suitable to use `InvalidArguments` status, much more like `Unexpected` or `Internal` Error.